src/hex/coordinate.rs

```diff
@@ -202,6 +354,148 @@ impl Axial {
+    ///
+    /// axial!(0, 0).vertices().iter().map(|vert| {/* ... */} );
+    ///
+    /// ```
+    pub fn vertices(&self) -> [Vertex; 6] {
+        [
+            vertex!(self.q, self.r, VertexSpin::Up),
+            vertex!(self.q + 1, self.r - 1, VertexSpin::Down),
+            vertex!(self.q, self.r + 1, VertexSpin::Up),
+            vertex!(self.q, self.r, VertexSpin::Down),
+            vertex!(self.q - 1, self.r + 1, VertexSpin::Up),
+            vertex!(self.q, self.r - 1, VertexSpin::Down),
+        ]
+    }
+
+    /// Given two neighboring tiles produce the shared vertices.
+    ///
+    /// See [`Vertex`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial, Vertex};
+    ///
+    /// let verts = axial!(0, 0).shared_vert_two(axial!(1, 0));
+    ///
+    /// ```
+    pub fn shared_vert_two(&self, b: Self) -> Option<[Vertex; 2]> {
+        for i in 0..=5 {
+            let dir = HexDirection::from(i);
+            let vec = dir.to_movement_vector();
+            if b == vec + *self {
+                // We found the neighbor and its direction.
+                // Front is in the positive q direction here.
+                return match dir {
+                    HexDirection::Front => Some([
+                        vertex!(self.q + 1, self.q - 1, VertexSpin::Down),
+                        vertex!(self.q, self.q + 1, VertexSpin::Up),
+                    ]),
+                    HexDirection::FrontRight => Some([
+                        vertex!(self.q, self.q + 1, VertexSpin::Up),
+                        vertex!(self.q, self.q, VertexSpin::Down),
+                    ]),
+                    HexDirection::BackRight => Some([
+                        vertex!(self.q, self.q, VertexSpin::Down),
+                        vertex!(self.q - 1, self.q + 1, VertexSpin::Up),
+                    ]),
+                    HexDirection::Back => Some([
+                        vertex!(self.q - 1, self.q + 1, VertexSpin::Up),
+                        vertex!(self.q, self.q - 1, VertexSpin::Down),
+                    ]),
+                    HexDirection::BackLeft => Some([
+                        vertex!(self.q, self.q - 1, VertexSpin::Down),
+                        vertex!(self.q, self.q, VertexSpin::Up),
+                    ]),
+                    HexDirection::FrontLeft => Some([
+                        vertex!(self.q, self.q, VertexSpin::Up),
+                        vertex!(self.q + 1, self.q - 1, VertexSpin::Down),
+                    ]),
+                };
+            }
+        }
+        None
+    }
+
+    /// Given three neighboring tiles produce the shared vertex.
+    ///
+    /// See [`Vertex`].
+    ///
+    /// # Example
+    /// ```
+    /// use gridava::hex::coordinate::{Axial, axial, Vertex};
+    ///
+    /// let verts = axial!(0, 0).shared_vert_three(axial!(1, 0), axial!(0, 1));
+    ///
+    /// ```
+    pub fn shared_vert_three(&self, b: Self, c: Self) -> Option<Vertex> {
```

Comment:
what should this do in a case where the three tiles are not neighboring?

Reply:
It will return none. All three must be neighboring for it to produce something.